server/src/main/java/io/unitycatalog/server/persist/TableRepository.java

```diff
@@ -241,32 +248,31 @@ public ListTablesResponse listTables(String catalogName,
                 List<TableInfoDAO> tableInfoDAOList = query.list();
                 returnNextPageToken = getNextPageToken(tableInfoDAOList);
                 for (TableInfoDAO tableInfoDAO : tableInfoDAOList) {
-                    TableInfo tableInfo = TableInfoConverter.convertToDTO(tableInfoDAO);
-                    if (!omitColumns) {
-                        tableInfo.setColumns(TableInfoConverter.convertColumnsToDTO(tableInfoDAO.getColumns()));
+                    TableInfo tableInfo = tableInfoDAO.toTableInfo();
+                    if (omitColumns) {
+                        tableInfo.setColumns(null);
```

Comment:
If omitColumns is true, we should not load the columns from ColumnDAO in the first place.

Reply:
Hibernate does this internally, we don't really control when it pulls columns, I can try to look and see if there is an option

Reply:
I implemented this, however, the openapi def does not have these properties so there is no way to write a test case right now.